cmd/opampsupervisor/supervisor/config/config.go

```diff
@@ -5,8 +5,22 @@ package config
 
 // Supervisor is the Supervisor config file format.
 type Supervisor struct {
-	Server *OpAMPServer
-	Agent  *Agent
+	Server       *OpAMPServer
+	Agent        *Agent
+	Capabilities *Capabilities `mapstructure:"capabilities"`
+}
+
+// Capabilities is the set of capabilities that the Supervisor supports.
+// TODO: Support more capabilities
+type Capabilities struct {
+	AcceptsRemoteConfig            *bool `mapstructure:"accepts_remote_config"`
+	AcceptsOpAMPConnectionSettings *bool `mapstructure:"accepts_opamp_connection_settings"`
```

Comment:
Looking at the implementation I don't think we support this capability at the moment.

Reply:
I probably misunderstood your original comment. I have kept the capabilities that have TODO tasks to complete and discarded the ones that have nothing. If you believe that we require the implementation of those features, I can convert this into a draft and reopen it once they are finished.